File: sdks/typescript/types/src/enums.ts
/**
 * Enum union types and runtime constants for ModulaCMS.
 *
 * @module enums
 */

/**
 * Lifecycle status of a content item.
 *
 * - `'draft'` - Work in progress, not publicly visible.
 * - `'published'` - Live and publicly accessible.
 */
export type ContentStatus = 'draft' | 'published'

/**
 * Supported field types for content schema definitions.
 * Determines the editor widget and validation rules applied to a field.
 */
export type FieldType =
  | 'text'
  | 'textarea'
  | 'number'
  | 'date'
  | 'datetime'
  | 'boolean'
  | 'select'
  | 'media'
  | '_id'
  | 'json'
  | 'richtext'
  | 'slug'
  | 'email'
  | 'url'

/**
 * Classifies how a route maps incoming URL patterns to content or behavior.
 *
 * - `'static'` - Fixed URL path mapped to a single content item.
 * - `'dynamic'` - URL pattern with parameters resolved at request time.
 * - `'api'` - Custom API endpoint, typically registered by a plugin.
 * - `'redirect'` - Redirect to another URL (returns metadata, not a 301).
 */
export type RouteType = 'static' | 'dynamic' | 'api' | 'redirect'

/**
 * All supported content output format identifiers.
 * Use this array for runtime validation of format values.
 *
 * @example
 * if (CONTENT_FORMATS.includes(userInput)) { ... }
 */
export const CONTENT_FORMATS = ["contentful", "sanity", "strapi", "wordpress", "clean", "raw"] as const;

/**
 * A content output format accepted by the `?format=` query parameter.
 * Derived from {@link CONTENT_FORMATS} to keep the runtime array and type in sync.
 */
export type ContentFormat = (typeof CONTENT_FORMATS)[number];

File: sdks/swift/Sources/Modula/Enums.swift
// MARK: - ContentStatus

public enum ContentStatus: String, Codable, Sendable {
    case draft
    case published
}

// MARK: - FieldType

public enum FieldType: String, Codable, Sendable {
    case text
    case textarea
    case number
    case date
    case datetime
    case boolean
    case select
    case media
    case id = "_id"
    case json
    case richtext
    case slug
    case email
    case url
    case title = "_title"
}

// MARK: - RouteType

public enum RouteType: String, Codable, Sendable {
    case `static`
    case dynamic
    case api
    case redirect
}

// MARK: - PluginStatus

public enum PluginStatus: String, Codable, Sendable {
    case installed
    case enabled
}

// MARK: - Operation

public enum Operation: String, Codable, Sendable {
    case insert = "INSERT"
    case update = "UPDATE"
    case delete = "DELETE"
}

// MARK: - Action

public enum Action: String, Codable, Sendable {
    case create
    case update
    case delete
    case publish
}

// MARK: - ConflictPolicy

public enum ConflictPolicy: String, Codable, Sendable {
    case lww
    case manual
}

// MARK: - BackupType

public enum BackupType: String, Codable, Sendable {
    case full
    case incremental
    case differential
}

// MARK: - BackupStatus

public enum BackupStatus: String, Codable, Sendable {
    case pending
    case inProgress = "in_progress"
    case completed
    case failed
}

// MARK: - VerificationStatus

public enum VerificationStatus: String, Codable, Sendable {
    case pending
    case verified
    case failed
}

// MARK: - BackupSetStatus

public enum BackupSetStatus: String, Codable, Sendable {
    case pending
    case complete
    case partial
}
